doc/changelog.rst

```diff
@@ -1,6 +1,7 @@
 Changelog
 =========
 
+* :bug:`516 major` Fire test_interrupt earlier & setting session as interrupted as well
```

Comment:
Fire test_interrupt earlier and properly mark session as interrupted when a test is interrupted